R/endpoints.R

```diff
@@ -1,22 +1,63 @@
+#' @title Shared Documentation for epidatr Parameters
+#'
+#' @description This is a central text for parameter documentation
+#'
+#' @name .epidatr_shared_params
+#' @keywords internal
+#'
+#' @param auth string. Your restricted access key (not the same as API key).
+#' @param locations character. Vector of locations to fetch.
+#' @param states character. Two letter state abbreviations.
+#' @param regions character. Vector of regions to fetch.
+#' @param epiweeks [`timeset`]. Epiweeks to fetch. Supports
+#'   [`epirange()`] and defaults to all ("*") dates.
+#' @param time_type string. The temporal resolution of the data (either "day" or
+#'  "week", depending on signal).
+#' @param names character. Names to fetch.
+#'   [`epirange()`] and defaults to all ("*") dates.
+#' @param dates [`timeset`]. Dates to fetch. Supports
+#'   [`epirange()`] and defaults to all ("*") dates.
+#' @param time_values [`timeset`]. Dates or epiweeks to fetch.
+#'   Supports [`epirange()`] and defaults to all ("*") dates.
+#' @param as_of Date. Optionally, the as-of date for the issues to fetch.
+#'   See the "Data Versioning" section for details.
+#' @param issues [`timeset`]. Optionally, the issue(s) of the
+#'   data to fetch. See the "Data Versioning" section for details.
+#' @param lag integer. Optionally, the lag of the issues to fetch.
+#'   See the "Data Versioning" section for details.
+#' @param fetch_args [`fetch_args`]. Additional arguments to pass
+#'   to `fetch()`. See `fetch_args_list()` for details.
+#' @param ... not used for values, forces later arguments to bind by name
+#'
+#' @section Data Versioning:
+#' Several endpoints support retrieving historical versions of the data.
+#' The following parameters control this and are mutually exclusive (only
+#' one can be provided at a time).
+#' \itemize{
+#'   \item \code{as_of}: (Date) Retrieve the data as it was on this date.
```

Comment:
Check format. `as_of` should be able to have same format (int, epirange, etc) as other date fields